pymc_marketing/mmm/transformers.py

```diff
@@ -914,6 +914,31 @@ def michaelis_menten(
     return alpha * x / (lam + x)
 
 
+def michaelis_menten(
```

Comment:
Do we need two functions?

Reply:
Can all of the `SaturationTransformations` be wrapped in as_tensor_variable instead of making these changes?

Reply:
> Do we need two functions?

We don't need them, but made my life easier to avoid the pesky [L70](https://github.com/pymc-labs/pymc-marketing/blob/f70db336ebf8766e7792279ca4148261f6b18edf/pymc_marketing/mmm/utils.py#L70)

> Can all of the `SaturationTransformations` be wrapped in as_tensor_variable instead of making these changes?
Do you have anything in mind? I can think of 2 ways:
* Adding boilerplate to all the classes. Perhaps defining a decorator to reduce boilerplate. 
* Rely on `Transformation._checks()` but here we would need to trust the type hints to use something like `signature(class_function).return_annotation` and wrap `pt.as_tensor_variable` if needed.

Perhaps you have something less convoluted in mind?

Reply:
Can you not add it in the one location of the `apply` method. That is a common method that is not overwritten by the subclasses

Reply:
I think having it for apply would make sure that other custom saturation transformations will not encounter this found bug as well

Reply:
It shouldn't hurt to has pt.as_tensor_variable wrapping an already TensorVariable. pytensor will figure out an optimization

Reply:
> Can you not add it in the one location of the `apply` method? (where I linked) That is a common method that is not overwritten by the subclasses

The linked `apply` method needs to be called within a model context. Not the case in the plotting function.

Reply:
The proposed `sample_curve` might work though, since it open a model context. Having a look

Reply:
Do we want to add another pm.Deterministic to the model? 
It would be the case if we use `_sample_curve`

Reply:
We wouldnt want to use the private methods of the saturation transformation. Does sample_curve with fit_result not get us close to the data that is plotted? (Just not scaled on x and y)